corehq/apps/reports/v2/filters/xpath_column.py

```diff
@@ -104,9 +104,14 @@ class NumericXpathColumnFilter(BaseXpathColumnFilter):
     @staticmethod
     def _get_number(value):
         try:
-            return int(value)
+            value = int(value)
         except ValueError:
-            return float(value)
+            value = float(value)
+
+        if value < 0:
+            value = "'{}'".format(value)
```

Comment:
Can you explain how this works?

Reply:
it inserts `-1` as a string in the expression `foo <= -1` so that it becomes `foo <= '-1'`, otherwise the parser treats `-1` as a `UnaryExpression` with `-` as the `op` and `1` as `right` and then throws an error when doing the comparison. also `foo = -1` throws an error in the parser as well

Reply:
I think we want to solve this in the parser, not here, since there are lots of places where this issue could crop up. I made a PR for that here: https://github.com/dimagi/commcare-hq/pull/24736, let me know what you think!

Reply:
@proteusvacuum merged this one in. Will update my implementation in the next PR and remove this commit.